doc/src/documentation_style_guide.rst

```diff
@@ -602,6 +633,8 @@ formatting guidelines should be followed when writing docstrings:
 * The various methods on the class can be mentioned in the docstring or
   examples if they are important, but details on them should go in the
   docstring for the method itself.
+* Be aware that :: creates code blocks, which should not be used in the
+  docstrings.
```

Comment:
There are some legitimate uses for them. People just need to be aware that this is what this does. If something is simply an indented paragraph, that is not a code block. But for example, some docstrings have ASCII art and those have to be put in code blocks to render properly. But any code example with example Python should be put in a doctest. Also, as with everything else, you should always check the final version as rendered by Sphinx to make sure it looks correct in the HTML.

Reply:
Sounds good, I've added some clarification here.